internal/controller/mytype/mytype.go

```diff
@@ -39,8 +38,7 @@ const (
 	errNotMyType    = "managed resource is not a MyType custom resource"
 	errTrackPCUsage = "cannot track ProviderConfig usage"
 	errGetPC        = "cannot get ProviderConfig"
-	errNoSecretRef  = "ProviderConfig does not reference a credentials Secret"
-	errGetSecret    = "cannot get credentials Secret"
+	errGetCreds     = "cannot get credentials"
```

Comment:
@negz would you mind also adding the rate limiter in the setup below and at controller manager creation? Here is an example https://github.com/crossplane/provider-gcp/pull/306

Reply:
Ah I forgot that this needed some plumbing - done.